### PR TITLE
Modify Task timestamps for GitHub sync

### LINKED ISSUE #888: Modify Task timestamps to accommodate github sync
- Tasks can be multi-sourced. They can either by created or modified from Code Corps or GitHub (and in the future from other integrations, too).
- We will add a `created_at` and `modified_at` timestamp to `Task`.
  - `created_at` refers to the timestamp it was created _on the source that created it_. This is not simply when our API received it or when it wrote to the database.
  - `modified_at` refers to the timestamp it was modified _on the source that modified it_.
  - Whenever we're doing an update, *last write wins*.
- We will add a `created_from` and `modified_from` field that can be either `github` or `code_corps` for now. This tracks the original source of the creation or modification event.

### PR DESCRIPTION
**Pull request by joshsmith**
# What's in this PR?

Currently a work in progress. Could use feedback on it so far, though.

## References
Fixes #888

### CODE REVIEW

File: lib/code_corps/github/adapters/task.ex [hunk 1]
```diff
@@ -26,6 +28,8 @@ defmodule CodeCorps.GitHub.Adapters.Task do
     task
     |> Map.from_struct
     |> MapTransformer.transform_inverse(@mapping)
+    |> Map.delete("created_at")
     |> Map.delete("number")
+    |> Map.delete("updated_at")
```
Comment (begedin) [inline lib/code_corps/github/adapters/task.ex]: Maybe a function `clear_autogenerated_github_keys`, which removes all of these keys, for added explicitness and self-documenting?

**Review by begedin — DISMISSED** [review]
A minor suggestion for better self-documentation of the adapter code, but past that point, I think this is good to go.

**Comment by joshsmith** [comment]
I'm not sure this is fully ready but not sure what's missing. Doesn't look like a test for changeset builder and not sure what else.

**Comment by begedin** [comment]
I added a couple of missing comments and cleaned up a few minor details. I think this is good to merge.

**Review by begedin — DISMISSED** [review]

**Pull request merged**